capten/agent/internal/api/plugin_cloud_provider_apis.go

```diff
@@ -23,6 +23,21 @@ func (a *Agent) AddCloudProvider(ctx context.Context, request *captenpluginspb.A
 
 	a.log.Infof("Add Cloud Provider %s request received", request.CloudType)
 
+	project, err := a.as.GetCloudProviderByCloudType(request.CloudType)
```

Comment:
These changes expected for Crossplane providers not cloud providers.

Do this for add cross plane provider and update crossplane providers